router/batchrouter/asyncdestinationmanager/bing-ads/bulk_uploader.go

```diff
@@ -0,0 +1,293 @@
+			FailedCount:   len(asyncDestStruct.FailedJobIDs) + len(asyncDestStruct.ImportingJobIDs),
+			DestinationID: destination.ID,
+		}
+	}
+	uploadRetryableStat := stats.Default.NewTaggedStat("events_over_prescribed_limit", stats.CountType, map[string]string{
+		"module":   "batch_router",
+		"destType": b.destName,
+	})
+	for _, actionFile := range actionFiles {
+		uploadRetryableStat.Count(len(actionFile.FailedJobIDs))
+		_, err := os.Stat(actionFile.ZipFilePath)
+		if err != nil || actionFile.EventCount == 0 {
+			continue
+		}
+		urlResp, err := b.service.GetBulkUploadUrl()
+		if err != nil {
+			b.logger.Error("Error in getting bulk upload url: %w", err)
+			failedJobs = append(append(failedJobs, actionFile.SuccessfulJobIDs...), actionFile.FailedJobIDs...)
+			errors = append(errors, fmt.Sprintf("%s:error in getting bulk upload url: %s", actionFile.Action, err.Error()))
+			continue
+		}
+
+		if urlResp.UploadUrl == "" || urlResp.RequestId == "" {
+			b.logger.Error(`{"error" : "getting empty string in upload url or request id. Provide correct configuration credentials."}`)
+			failedJobs = append(append(failedJobs, actionFile.SuccessfulJobIDs...), actionFile.FailedJobIDs...)
+			errors = append(errors, fmt.Sprintf("%s:getting empty string in upload url or request id. Provide correct configuration credentials.", actionFile.Action))
+			continue
+		}
+
+		uploadTimeStat := stats.Default.NewTaggedStat("async_upload_time", stats.TimerType, map[string]string{
+			"module":   "batch_router",
+			"destType": b.destName,
+		})
+
+		startTime := time.Now()
+		uploadBulkFileResp, errorDuringUpload := b.service.UploadBulkFile(urlResp.UploadUrl, actionFile.ZipFilePath)
+		uploadTimeStat.Since(startTime)
+
+		if errorDuringUpload != nil {
+			b.logger.Error("error in uploading the bulk file: %v", errorDuringUpload)
+			failedJobs = append(append(failedJobs, actionFile.SuccessfulJobIDs...), actionFile.FailedJobIDs...)
+			errors = append(errors, fmt.Sprintf("%s:error in uploading the bulk file: %v", actionFile.Action, errorDuringUpload))
+			continue
+		}
+
+		if uploadBulkFileResp.RequestId == "" || uploadBulkFileResp.TrackingId == "" {
+			failedJobs = append(append(failedJobs, actionFile.SuccessfulJobIDs...), actionFile.FailedJobIDs...)
+			errors = append(errors, fmt.Sprintf("%s:getting empty string in upload url or request id, Provide correct configuration credentials.", actionFile.Action))
+			continue
+		}
+		importIds = append(importIds, uploadBulkFileResp.RequestId)
+		failedJobs = append(failedJobs, actionFile.FailedJobIDs...)
+		successJobs = append(successJobs, actionFile.SuccessfulJobIDs...)
+	}
+
+	var parameters common.ImportParameters
+	parameters.ImportId = strings.Join(importIds, commaSeparator)
+	importParameters, err := stdjson.Marshal(parameters)
+	if err != nil {
+		b.logger.Error("Errored in Marshalling parameters" + err.Error())
+	}
+	allErrors := `{"error":"` + strings.Join(errors, commaSeparator) + `"}`
```

Comment:
we could use the `EnhanceJSON` method present in `router/utils` for all such error messages

Reply:
done